src/lib.rs

```diff
@@ -279,6 +282,12 @@ impl UnixStream {
         self.io.get_ref().peer_addr()
     }
 
+    /// Returns effective credentials of the process which called `connect` or `socketpair`.
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    pub fn peer_cred(&self) -> io::Result<UCred> {
```

Comment:
Digging into the topic more I found that:

* On Linux, `getsockopt` should be used
* BSD families (including macOS) use `getpeereid`
* Solaris uses `getpeerucred`

I know of no other unix used in the wild with Rust.

So maybe we can just remove `#[cfg]`. However, it'd be nice to implement this for Solaris too. But as I said, I have nothing to test it on.

Also, `macos` cfg will have to be changed to `#[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd"))]`

What do you think?